Include/internal/pycore_object.h

```diff
@@ -613,6 +613,20 @@ _PyObject_IS_GC(PyObject *obj)
             && (type->tp_is_gc == NULL || type->tp_is_gc(obj)));
 }
 
+// Fast inlined version of PyObject_Hash()
+static inline Py_hash_t
+_PyObject_HashFast(PyObject *op) {
+    Py_hash_t hash;
+    if (!PyUnicode_CheckExact(op) ||
+        (hash = FT_ATOMIC_LOAD_SSIZE_RELAXED(_PyASCIIObject_CAST(op)->hash)) == -1) {
+        hash = PyObject_Hash(op);
+        if (hash == -1) {
+            return -1;
+        }
```

Comment:
Now that this is its own function, it would probably be more clear to refactor it a bit like:

```
if (PyUnicode_CheckExact(op)) {
   Py_hash_t hash = ...;
   if (hash != -1) {
       return hash;
   }
}
return PyObject_Hash(op);
```

I think that would make it a bit more clear that the unicode case is handled specially and the other cases just call `PyObject_Hash`.

Reply:
Thaks for review, updated!